dagger-reflect/reflect/src/main/java/dagger/reflect/ReflectiveMembersInjector.java

```diff
@@ -33,11 +36,11 @@
   static <T> MembersInjector<T> create(Class<T> cls, BindingGraph graph) {
     // TODO throw if interface?
 
-    Map<Field, Binding<?>> fieldBindings = new LinkedHashMap<>();
-    Map<Method, Binding<?>[]> methodBindings = new LinkedHashMap<>();
-
+    List<Bindings> hierarchyBindings = new LinkedList<>();
```

Comment:
chose linked over array since most of the time this will be less than 10 (default array capacity)